packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -0,0 +1,326 @@
+		const dataObject2 = await requestFluidObject<ITestFluidObject>(container2, "default");
+		sharedMap2 = await dataObject2.getSharedObject<SharedMap>(mapId);
+
+		const container3 = await provider.loadTestContainer(testContainerConfig);
+		const dataObject3 = await requestFluidObject<ITestFluidObject>(container3, "default");
+		sharedMap3 = await dataObject3.getSharedObject<SharedMap>(mapId);
+
+		sharedMap1.set("testKey1", "testValue");
+
+		await provider.ensureSynchronized();
+	});
+
+	it("has no compressor if not enabled", async () => {
+		provider.reset();
+		const config: ITestContainerConfig = {
+			registry: [
+				[mapId, SharedMap.getFactory()],
+				[cellId, SharedCell.getFactory()],
+			],
+			fluidDataObjectType: DataObjectFactoryType.Test,
+		};
+		const container1 = await provider.makeTestContainer(config);
+		const dataObject1 = await requestFluidObject<ITestFluidObject>(container1, "default");
+		const map = await dataObject1.getSharedObject<SharedMap>(mapId);
+
+		assert(getIdCompressor(map) === undefined);
+	});
+
+	it("can normalize session space IDs to op space", async () => {
+		assert(getIdCompressor(sharedMap1) !== undefined);
+		assert(getIdCompressor(sharedMap2) !== undefined);
+		assert(getIdCompressor(sharedMap3) !== undefined);
+
+		const compressedIds: Map<SessionSpaceCompressedId, SharedMap> = new Map();
+		for (let i = 0; i < 1024; i++) {
+			compressedIds.set(getIdCompressor(sharedMap1).generateCompressedId(), sharedMap1);
+			compressedIds.set(getIdCompressor(sharedMap2).generateCompressedId(), sharedMap1);
+			compressedIds.set(getIdCompressor(sharedMap3).generateCompressedId(), sharedMap1);
+		}
+
+		for (const [id, map] of compressedIds) {
+			assert(getIdCompressor(map) !== undefined);
+			assert(getIdCompressor(map).normalizeToOpSpace(id) < 0);
+		}
+
+		// Generate DDS ops so that the compressor synchronize
+		sharedMap1.set("key", "value");
+		sharedMap2.set("key2", "value2");
+		sharedMap3.set("key3", "value3");
+
+		await provider.ensureSynchronized();
+
+		for (const [id, map] of compressedIds) {
+			assert(getIdCompressor(map) !== undefined);
+			assert(getIdCompressor(map).normalizeToOpSpace(id) >= 0);
+		}
+
+		assert.strictEqual(sharedMap1.get("key"), "value");
+		assert.strictEqual(sharedMap2.get("key2"), "value2");
+		assert.strictEqual(sharedMap3.get("key3"), "value3");
+	});
+
+	it("can normalize local op space IDs from a local session to session space", async () => {
+		assert(getIdCompressor(sharedMap1) !== undefined);
+		const sessionSpaceId = getIdCompressor(sharedMap1).generateCompressedId();
+		sharedMap1.set("key", "value");
+
+		await provider.ensureSynchronized();
+		const opSpaceId = getIdCompressor(sharedMap1).normalizeToOpSpace(sessionSpaceId);
+		const normalizedSessionSpaceId = getIdCompressor(sharedMap1).normalizeToSessionSpace(
+			opSpaceId,
+			getIdCompressor(sharedMap1).localSessionId,
+		);
+
+		assert(opSpaceId >= 0);
```

Comment:
Moving this assert next to opSpaceId assignment might be better as you don't need to do the next step if it fails.